src/include/brain/tf_session_entity/tf_session_entity.h

```diff
@@ -0,0 +1,90 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// tf_session_entity.h
+//
+// Identification: src/include/brain/tf_session_entity/tf_session_entity.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "brain/tf_session_entity/tf_session_entity_input.h"
+#include "brain/tf_session_entity/tf_session_entity_output.h"
+
+#define TFSE_TEMPLATE_ARGUMENTS \
+  template <typename InputType, typename OutputType>
+#define TFSE_TYPE TfSessionEntity<InputType, OutputType>
+
+namespace peloton {
+namespace brain {
+
+/**
+ * The `TfSessionEntity` class is the main entity class for
+ * handling Tensorflow sessions and evaluations within.
+ * Since creating a graph in Tensorflow C API is pretty challenging
+ * it is recommended to generally create a graph in a more stable API(eg.
+ * python)
+ * and import its serialized(.pb) version for use here.
+ * This class supports and makes the following simplified:
+ * (1) Session initialization
+ * (2) Graph Imports: Using `ImportGraph`
+ * (3) Evaluation of Ops: using different versions of the `Eval` fn.
+ * The inputs and outputs are simplified by using the wrapper classes
+ * `TfSessionEntityInput`, `TfSessionEntityOutput`.
+ * (4) Misc Helper operations
+ */
+TFSE_TEMPLATE_ARGUMENTS
+class TfSessionEntity {
+ public:
+  TfSessionEntity();
+  ~TfSessionEntity();
+
+  // Graph Import
+  void ImportGraph(const std::string &filename);
+
+  /**
+   * Evaluation/Session.Run() options.
+   * Provides a set of options to evaluate an Op either with
+   * 1. No input/outputs eg. Global variable initialization
+   * 2. Inputs, but no outputs eg. Backpropagation
+   * 3. Inputs and Outputs eg. Loss calculation, Predictions.
+   */
+  void Eval(const std::string &opName);
+  void Eval(std::vector<TfSessionEntityInput<InputType>> helper_inputs,
+            const std::string &op_name);
+  OutputType *Eval(
+      std::vector<TfSessionEntityInput<InputType>> helper_inputs,
+      std::vector<TfSessionEntityOutput<OutputType>> helper_outputs);
```

Comment:
Const reference. If we replace all these references, I think we can also remove the includes for the input and output types, and move them to the .cpp file.

Reply:
I've fixed everything else. I'm not sure what you mean here - Move the includes from line 15 and 16 to the .cpp?

Reply:
Yes. I think that's doable if you replace the vector arguments here with their const references.

Reply:
Then I'll have to import `#include <vector>` and similarly for string. 

Also, I thought the convention was to keep includes in the .h files only.